subsys/bluetooth/host/hci_core.c

```diff
@@ -830,8 +830,8 @@ static int le_adv_set_private_addr(struct bt_le_ext_adv *adv)
 	bt_addr_t nrpa;
 	int err;
 
-	if (IS_ENABLED(CONFIG_BT_EXT_ADV) &&
-	    BT_FEAT_LE_EXT_ADV(bt_dev.le.features)) {
+	if (!(IS_ENABLED(CONFIG_BT_EXT_ADV) &&
+	      BT_FEAT_LE_EXT_ADV(bt_dev.le.features))) {
 		return le_set_private_addr(adv->id);
```

Comment:
The creation of the address value seems to be exactly the same in `le_set_private_addr()` (above this function) and what happens in the remainder of this function, so would it not make sense to keep that part here and then either call `set_random_addr(&nrpa);` in the case of this branch or `set_adv_random_address(adv, &nrpa);` in the else-branch? That would reduce the code a little bit.

Reply:
`set_adv_random_address` will call `set_random_address` if needed, so calling it always should handle it.